src/DeviceLog.h

```diff
@@ -16,81 +16,116 @@ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLI
+
+    virtual bool OpenWithSystemApp();
+
+    virtual bool SaveData(p_data_hdr_t *dataHdr, const uint8_t *dataBuf, protocol_type_t ptype = _PTYPE_INERTIAL_SENSE_DATA);
+
+    virtual bool SaveData(int dataSize, const uint8_t *dataBuf, cLogStats &globalLogStats);
+
+    virtual p_data_buf_t *ReadData() = 0;
+
+    virtual void SetSerialNumber(uint32_t serialNumber) = 0;
+
+
+    virtual std::string LogFileExtention() = 0;
+
+    virtual void Flush() {}
+
+    bool SetupReadInfo(const std::string &directory, const std::string &deviceName, const std::string &timeStamp);
+
+    ISDevice* Device();
+
+    const dev_info_t *DeviceInfo();
+
+    uint16_t HardwareId() { return devHdwId; }
+    uint32_t SerialNumber() { return devSerialNo; }
+
+    // void SetDeviceInfo(const dev_info_t *info);
+    uint64_t FileSize() { return m_fileSize; }
+
+    uint64_t LogSize() { return m_logSize; }
+
+    uint32_t FileCount() { return m_fileCount; }
+
+    std::string GetNewFileName(uint32_t serialNumber, uint32_t fileCount, const char *suffix);
+
+    void SetKmlConfig(bool gpsData = true, bool showTracks = true, bool showPoints = true, bool showPointTimestamps = true, double pointUpdatePeriodSec = 1.0, bool altClampToGround = true) {
+        m_enableGpsLogging = gpsData;
+        m_showTracks = showTracks;
+        m_showPoints = showPoints;
+        m_showPointTimestamps = showPointTimestamps;
+        m_pointUpdatePeriodSec = pointUpdatePeriodSec;
+        m_altClampToGround = altClampToGround;
+    }
+
+    void ShowParseErrors(bool show) { m_showParseErrors = show; }
 
 protected:
-	bool OpenNewSaveFile();
-	bool OpenNextReadFile();
-    void OnReadData(p_data_buf_t* data);
-
-	std::vector<std::string> m_fileNames;
-	cISLogFileBase*         m_pFile;
-	std::string             m_directory;
-	std::string             m_timeStamp;
-	std::string             m_fileName;
-	bool                    m_writeMode;	// Logger initialized for writing
-	bool					m_showParseErrors;
-	dev_info_t              m_devInfo;
-	int                     m_pHandle;
-	uint64_t                m_fileSize;
-	uint64_t                m_logSize;
-	uint32_t                m_fileCount;
-	uint64_t                m_maxDiskSpace;
-	uint32_t                m_maxFileSize;
-	bool                    m_altClampToGround;
-	bool                    m_enableGpsLogging;
-	bool                    m_showTracks;
-	bool                    m_showPoints;
-	bool                    m_showPointTimestamps;
-	double                  m_pointUpdatePeriodSec;
+    bool OpenNewSaveFile();
+
+    bool OpenNextReadFile();
+
+    void OnReadData(p_data_buf_t *data);
+
+    const ISDevice *device = nullptr;               //! ISDevice reference to source of data
+
+    uint16_t devHdwId = 0;                          //! used when reading a file and no ISDevice is available
+    uint32_t devSerialNo = -1;                      //! used when reading a file, and no ISDevice is available
```

Comment:
I'm a bit of a fan of keeping uniform convention.  All of the other member variables have prefix "m_".  It would be nice if these did also.